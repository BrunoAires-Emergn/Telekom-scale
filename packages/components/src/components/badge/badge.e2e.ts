import { newE2EPage } from '@stencil/core/testing';

describe('scale-badge', () => {
  it('renders', async () => {
    const page = await newE2EPage();
    await page.setContent('<scale-badge>Click me!</scale-badge>');
    const element = await page.find('scale-badge');
    expect(element).toHaveClass('hydrated');
  });
});
