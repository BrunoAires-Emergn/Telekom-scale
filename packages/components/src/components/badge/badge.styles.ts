import { JssStyle } from 'jss';

export const styles: JssStyle = {
  badge: {
    display: 'inline-block',
    padding: '0.25em 0.4em',
    fontSize: '90%',
    fontWeight: 700,
    lineHeight: 1,
    textAlign: 'center',
    whiteSpace: 'nowrap',
    verticalAlign: 'baseline',
    borderRadius: '0.25rem',
    transition: 'all 0.15s ease-in-out',
    background: '#343a40',
    color: ' #fff',
  },
  'badge--pill': {
    paddingRight: '0.6em',
    paddingLeft: '0.6em',
    borderRadius: '10rem',
  },
  'badge--variant-dark': {
    color: '#fff',
    backgroundColor: '#343a40',
  },
  'badge--variant-light': {
    color: '#212529',
    backgroundColor: '#f8f9fa',
  },
  'badge--variant-info': {
    color: '#fff',
    backgroundColor: '#17a2b8',
  },
  'badge--variant-warning': {
    color: '#212529',
    backgroundColor: '#ffc107',
  },
  'badge--variant-danger': {
    color: '#fff',
    backgroundColor: '#dc3545',
  },
  'badge--variant-success': {
    color: '#fff',
    backgroundColor: '#28a745',
  },
  'badge--variant-secondary': {
    color: '#fff',
    backgroundColor: '#6c757d',
  },
  'badge--variant-primary': {
    color: '#fff',
    backgroundColor: '#007bff',
  },
  'badge--link': {
    textDecoration: 'none',
    '&:hover': {
      background: '#1d2124',
    },
    '&:active': {
      background: '#1d2124',
    },
    '&:focus': {
      outline: 0,
      boxShadow: '0 0 0 0.2rem rgba(52, 58, 64, 0.5)',
    },
  },
};
