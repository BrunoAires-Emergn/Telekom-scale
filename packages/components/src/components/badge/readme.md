# t-badge



<!-- Auto Generated Below -->


## Properties

| Property      | Attribute      | Description                    | Type                                     | Default     |
| ------------- | -------------- | ------------------------------ | ---------------------------------------- | ----------- |
| `customClass` | `custom-class` | (optional) Badge class         | `string`                                 | `''`        |
| `link`        | `link`         | (optional) Badge link          | `string`                                 | `''`        |
| `pill`        | `pill`         | (optional) Badge pill          | `boolean`                                | `false`     |
| `size`        | `size`         | (optional) Badge size          | `string`                                 | `''`        |
| `styles`      | --             | (optional) Injected jss styles | `StyleSheet<string \| number \| symbol>` | `undefined` |
| `variant`     | `variant`      | (optional) Badge variant       | `string`                                 | `''`        |


## Methods

### `updateStyles(newStyle: StyleSheet<string | number | symbol>) => Promise<void>`

Method: updateStyles()

#### Returns

Type: `Promise<void>`




----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
