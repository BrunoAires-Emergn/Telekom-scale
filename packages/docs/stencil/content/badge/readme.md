# t-badge



<!-- Auto Generated Below -->


## Properties

| Property      | Attribute      | Description                    | Type                                     | Default     |
| ------------- | -------------- | ------------------------------ | ---------------------------------------- | ----------- |
| `customClass` | `custom-class` | (optional) Badge class         | `string`                                 | `''`        |
| `link`        | `link`         | (optional) Badge link          | `string`                                 | `''`        |
| `pill`        | `pill`         | (optional) Badge pill          | `boolean`                                | `false`     |
| `size`        | `size`         | (optional) Badge size          | `string`                                 | `''`        |
| `styles`      | --             | (optional) Injected jss styles | `StyleSheet<string \| number \| symbol>` | `undefined` |
| `variant`     | `variant`      | (optional) Badge variant       | `string`                                 | `''`        |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
