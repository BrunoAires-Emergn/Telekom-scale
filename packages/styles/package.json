{
    "name": "@telements/styles",
    "version": "0.0.1",
    "main": "dist/index.js",
    "scripts": {
        "build": "tsc -b",
        "postbuild": "node ./dist/export-css.js",
        "setup": "yarn"
    },
    "author": "@eeegor <egor.kirpichev@telekom.de>",
    "license": "MIT",
    "devDependencies": {
        "typescript": "^3.4.5"
    },
    "dependencies": {

    }
}
