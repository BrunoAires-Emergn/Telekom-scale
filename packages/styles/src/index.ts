import { button } from './button';
import { alert } from './alert';
import { card } from './card';

const theme = {
	colors: {
		white: '#fff',
		black: '#000',
		blue: '#00f',
		red: '#f00',
	}
}

export {
	theme,
	alert,
	button,
	card,
}
