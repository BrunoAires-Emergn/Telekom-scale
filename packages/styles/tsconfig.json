{
	"extends": "../tsconfig.base",
	"compilerOptions": {
		"rootDir": "./src",
		"outDir": "./dist",
		"types": [
			"node",
			"jest"
		]
	}
}
