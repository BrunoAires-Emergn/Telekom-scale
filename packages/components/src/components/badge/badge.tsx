import { Component, Prop, h, Host, Method } from '@stencil/core';
import { CssClassMap } from '../../utils/utils';
import classNames from 'classnames';
import { styles } from './badge.styles';
import { CssInJs } from '../../utils/css-in-js';
import { StyleSheet } from 'jss';
import Base from '../../utils/base-interface';

@Component({
  tag: 't-badge',
  shadow: true,
})
export class Badge implements Base {
  /** (optional) Badge class */
  @Prop() customClass?: string = '';
  /** (optional) Badge size */
  @Prop() size?: string = '';
  /** (optional) Badge variant */
  @Prop() variant?: string = '';
  /** (optional) Badge pill */
  @Prop() pill?: boolean = false;
  /** (optional) Badge link */
  @Prop() link?: string = '';

  /** (optional) Injected jss styles */
  @Prop() styles?: StyleSheet;
  /** decorator Jss stylesheet */
  @CssInJs('Badge', styles) stylesheet: StyleSheet;

  componentWillLoad() {}
  componentWillUpdate() {}

  /** Method: updateStyles()  */
  @Method()
  async updateStyles(newStyle: StyleSheet) {
    this.styles = newStyle;
  }

  render() {
    return (
      <Host>
        <style>{this.stylesheet.toString()}</style>
        {!!this.link ? (
          <a href={this.link} class={this.getCssClassMap()}>
            <slot />
          </a>
        ) : (
          <span class={this.getCssClassMap()}>
            <slot />
          </span>
        )}
      </Host>
    );
  }

  getCssClassMap(): CssClassMap {
    const { classes } = this.stylesheet;
    return classNames(
      classes.badge,
      this.customClass && this.customClass,
      this.size && classes[`badge--size-${this.size}`],
      this.variant && classes[`badge--variant-${this.variant}`],
      this.pill && classes[`badge--pill`],
      !!this.link && classes[`badge--link`]
    );
  }
}
