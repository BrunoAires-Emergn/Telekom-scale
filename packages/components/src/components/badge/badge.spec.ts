import { newSpecPage } from '@stencil/core/testing';
import { Badge } from './badge';
import { styles } from './badge.styles';
import jss from 'jss';

describe('Badge', () => {
  let element;
  let stylesheet;
  beforeEach(async () => {
    element = new Badge();
    stylesheet = element.stylesheet = jss.createStyleSheet(styles as any);
  });

  it('should match snapshot', async () => {
    const page = await newSpecPage({
      components: [Badge],
      html: `<t-badge>Label</t-badge>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  // it('should have a default css class', () => {
  //   expect(element.getCssClassMap()).toBe('badge');
  // });

  it('should have a link', async () => {
    const page = await newSpecPage({
      components: [Badge],
      html: `<t-badge link="#">Label</t-badge>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  // it('should handle size css class', () => {
  //   element.size = 'small';
  //   expect(element.getCssClassMap()).toContain('badge--size-small');
  // });

  // it('should have class variant primary', () => {
  //   element.variant = 'primary';
  //   expect(element.getCssClassMap()).toContain('badge--variant-primary');
  // });

  // it('should render pill badge', () => {
  //   element.pill = true;
  //   expect(element.getCssClassMap()).toContain('badge--pill');
  // });
});
