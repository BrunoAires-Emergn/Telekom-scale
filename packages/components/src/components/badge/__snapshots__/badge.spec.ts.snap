// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge should have a link 1`] = `
<t-badge link="#">
  <mock:shadow-root>
    <style>
      .badge-0-0-34 {
  color:  #fff;
  display: inline-block;
  padding: 0.25em 0.4em;
  font-size: 90%;
  background: #343a40;
  text-align: center;
  transition: all 0.15s ease-in-out;
  font-weight: 700;
  line-height: 1;
  white-space: nowrap;
  border-radius: 0.25rem;
  vertical-align: baseline;
}
.badge--pill-0-0-35 {
  padding-left: 0.6em;
  border-radius: 10rem;
  padding-right: 0.6em;
}
.badge--variant-dark-0-0-36 {
  color: #fff;
  background-color: #343a40;
}
.badge--variant-light-0-0-37 {
  color: #212529;
  background-color: #f8f9fa;
}
.badge--variant-info-0-0-38 {
  color: #fff;
  background-color: #17a2b8;
}
.badge--variant-warning-0-0-39 {
  color: #212529;
  background-color: #ffc107;
}
.badge--variant-danger-0-0-40 {
  color: #fff;
  background-color: #dc3545;
}
.badge--variant-success-0-0-41 {
  color: #fff;
  background-color: #28a745;
}
.badge--variant-secondary-0-0-42 {
  color: #fff;
  background-color: #6c757d;
}
.badge--variant-primary-0-0-43 {
  color: #fff;
  background-color: #007bff;
}
.badge--link-0-0-44 {
  text-decoration: none;
}
.badge--link-0-0-44:hover {
  background: #1d2124;
}
.badge--link-0-0-44:active {
  background: #1d2124;
}
.badge--link-0-0-44:focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(52, 58, 64, 0.5);
}
    </style>
    <a class="badge--link-0-0-44 badge-0-0-34" href="#">
      <slot></slot>
    </a>
  </mock:shadow-root>
  Label
</t-badge>
`;

exports[`Badge should match snapshot 1`] = `
<t-badge>
  <mock:shadow-root>
    <style>
      .badge-0-0-12 {
  color:  #fff;
  display: inline-block;
  padding: 0.25em 0.4em;
  font-size: 90%;
  background: #343a40;
  text-align: center;
  transition: all 0.15s ease-in-out;
  font-weight: 700;
  line-height: 1;
  white-space: nowrap;
  border-radius: 0.25rem;
  vertical-align: baseline;
}
.badge--pill-0-0-13 {
  padding-left: 0.6em;
  border-radius: 10rem;
  padding-right: 0.6em;
}
.badge--variant-dark-0-0-14 {
  color: #fff;
  background-color: #343a40;
}
.badge--variant-light-0-0-15 {
  color: #212529;
  background-color: #f8f9fa;
}
.badge--variant-info-0-0-16 {
  color: #fff;
  background-color: #17a2b8;
}
.badge--variant-warning-0-0-17 {
  color: #212529;
  background-color: #ffc107;
}
.badge--variant-danger-0-0-18 {
  color: #fff;
  background-color: #dc3545;
}
.badge--variant-success-0-0-19 {
  color: #fff;
  background-color: #28a745;
}
.badge--variant-secondary-0-0-20 {
  color: #fff;
  background-color: #6c757d;
}
.badge--variant-primary-0-0-21 {
  color: #fff;
  background-color: #007bff;
}
.badge--link-0-0-22 {
  text-decoration: none;
}
.badge--link-0-0-22:hover {
  background: #1d2124;
}
.badge--link-0-0-22:active {
  background: #1d2124;
}
.badge--link-0-0-22:focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(52, 58, 64, 0.5);
}
    </style>
    <span class="badge-0-0-12">
      <slot></slot>
    </span>
  </mock:shadow-root>
  Label
</t-badge>
`;
